modules/process/manager.go

```diff
@@ -111,30 +113,15 @@ func (pm *Manager) ExecDirEnv(timeout time.Duration, dir, desc string, env []str
 	}
 
 	pid := pm.Add(desc, cmd)
-	done := make(chan error)
-	go func() {
-		done <- cmd.Wait()
-	}()
-
-	var err error
-	select {
-	case <-time.After(timeout):
-		if errKill := pm.Kill(pid); errKill != nil {
-			log.Error(4, "exec(%d:%s) failed to kill: %v", pid, desc, errKill)
-		}
-		<-done
-		return "", "", ErrExecTimeout
-	case err = <-done:
-	}
-
+	err := cmd.Wait()
 	pm.Remove(pid)
 
-	if err != nil {
-		out := fmt.Errorf("exec(%d:%s) failed: %v stdout: %v stderr: %v", pid, desc, err, stdOut, stdErr)
-		return stdOut.String(), stdErr.String(), out
+	if err == nil {
```

Comment:
I'd change the error-handling to a more go-esque way
```go
if err != nil {
  err = fmt.Errorf(...) // NOTE: include the original error as well as the `ctx.Err()`...
}

return stdOut.String(), stdErr.String(), err
```

Reply:
@typeless maybe you could change this.

Reply:
Done.